### DISCUSSION TITLE
Integration into the npm ecosystem

### DISCUSSION THREAD

**Discussion by ilg-ul** [jothepro/doxygen-awesome-css#113] [Ideas]
I did a test with this theme, and I think it is a nice addition to Doxygen. Thank you!

Since my environment is npm centric, to simplify its use in multiple projects I did a fork and published the project on npmjs.com:

- <https://www.npmjs.com/package/@xpack-3rd-party/doxygen-awesome-css>

As documented in the README, it can be conveniently installed with npm/xpm.

The additions are minimal, basically a `package.json` (for the package identify) and a `.npmignore` to filter out files when publishing.

It would be great if you could add these files to the project and possibly publish the releases on npmjs.com too. This step is not mandatory, xpm/npm can install packages directly from GitHub, but from npmjs.com it is faster.

If you think useful, I can submit a PR with a proposal for the 2 files.

**Comment by jothepro** [comment]
I think it's a good idea to add convenient methods for installation. For example, @Tachi107  (https://github.com/jothepro/doxygen-awesome-css/pull/89) previously contributed a Makefile for distribution in Debian.

Your PR to add the `package.json` and `.npmignore` would be a great addition IMO!

I am not (yet) sure if I want to publish to npmjs, as this may come with additional maintenance overhead for me. Is it possible to automate the publishing process to npmjs in the github actions workflow?

**Comment by ilg-ul** [comment]
> Is it possible to automate the publishing process to npmjs in the github actions workflow?

I guess it is, according to the npm [documentation](https://docs.npmjs.com/creating-and-viewing-access-tokens) it should be possible to generate security tokens which provide publish rights, but personally I do not have experience with them.

I have tens of npm packages, but I still prefer to publish them manually from my computer, to be sure I do not mess anything.

The sequence is:

```
npm version X.Y.Z # to set the new version in package.json
npm pack # to preview the package content
npm publish # to actually publish on npmjs.com
```

I guess you can use `npm version` and `npm publish` in a GitHub Action (also passing the access token), but I think that this maintenance overhead is minimal, and the actual publishing can be done locally without problems.

**Reply by ilg-ul** [reply]
To be noted that `npm publish` also pushes a git tag with the new version, so you no longer have to do it manually.

**Comment by ilg-ul** [comment]
PR: https://github.com/jothepro/doxygen-awesome-css/pull/114

**Comment by jothepro** [comment]
PR #114 was merged. Closing the Discussion for now.

**Reply by ilg-ul** [reply]
Looking forward to the new release. Don't forget to update the version in package.json.
